core/src/main/java/com/oath/oak/Handle.java

```diff
@@ -196,27 +200,33 @@ public ByteOrder order() {
         return transformation;
     }
 
-    /**
-     * Applies a transformation under writers locking
-     *
-     * @param transformer transformation to apply
-     * @return Transformation result or null if value is deleted
-     */
-    <T> T mutatingTransform(Function<ByteBuffer, T> transformer) {
-        T result;
+    <V> V exchange(V newValue, Function<ByteBuffer, V> valueDeserializeTransformer, OakSerializer<V> serializer, MemoryManager memoryManager) {
```

Comment:
I like the idea of encapsulating atomic exchange, but I think there's subtle semantic difference that is lost here. Specifically, these implementation of exchange and compareExchange are not zero copy. The original idea behind `mutatingTransform` was to be a potentially zero-copy transformation that takes a write lock. The thought was that going forward we may want to extend the zero-copy API to support more standard API methods while keeping ZC performance.

Reply:
We can perhaps pass the chunk's comparator instead of valueDeserializeTransformer to support a ZC exchange.